planner/core/expression_rewriter.go

```diff
@@ -712,21 +715,35 @@ func (er *expressionRewriter) handleInSubquery(ctx context.Context, v *ast.Patte
 		er.err = expression.ErrOperandColumns.GenWithStackByArgs(lLen)
 		return v, true
 	}
-	var rexpr expression.Expression
-	if np.Schema().Len() == 1 {
-		rexpr = np.Schema().Columns[0]
-		rCol := rexpr.(*expression.Column)
-		// For AntiSemiJoin/LeftOuterSemiJoin/AntiLeftOuterSemiJoin, we cannot treat `in` expression as
-		// normal column equal condition, so we specially mark the inner operand here.
-		if v.Not || asScalar {
+	// For AntiSemiJoin/LeftOuterSemiJoin/AntiLeftOuterSemiJoin, we cannot treat `in` expression as
+	// normal column equal condition, so we specially mark the inner operand here.
+	if v.Not || asScalar {
+		for i, rCol := range np.Schema().Columns {
```

Comment:
This change also extends this method to handle `multi-column IN`? If so, we'd better add tests like `(1, null) in (...)`, `(a, null) in (...)` ....

Reply:
+1